presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -789,29 +799,60 @@ public void addUtilizedTableColumnReferences(AccessControlInfo accessControlInfo
         return ImmutableMap.copyOf(utilizedTableColumnReferences);
     }
 
-    public Map<AccessControlInfo, Map<QualifiedObjectName, Set<String>>> getTableColumnReferencesForAccessControl(Session session)
+    public Map<AccessControlInfo, Map<QualifiedObjectName, Set<Subfield>>> getTableColumnAndSubfieldReferencesForAccessControl(Session session)
     {
-        Map<AccessControlInfo, Map<QualifiedObjectName, Set<String>>> references = isCheckAccessControlOnUtilizedColumnsOnly(session) ? utilizedTableColumnReferences : tableColumnReferences;
+        Map<AccessControlInfo, Map<QualifiedObjectName, Set<Subfield>>> references;
+        if (!isCheckAccessControlWithSubfields(session)) {
+            references = (isCheckAccessControlOnUtilizedColumnsOnly(session) ? utilizedTableColumnReferences : tableColumnReferences)
+                    .entrySet()
+                    .stream()
```

Comment:
nit: put `.entrySet().stream()` at the end of the last line. We usually break a new line at `stream()`